server/entitymanager/entitymanager.go

```diff
@@ -57,14 +62,40 @@ type InMemoryEntityManager struct {
 }
 
 // ResolveChassis returns an entity based on the provided lookup.
-func (m *InMemoryEntityManager) ResolveChassis(chassis *service.EntityLookup) (*service.ChassisEntity, error) {
+// In cases when the serail for  moulder chassis is not set, it uses the controller card to find the chasiss.
+func (m *InMemoryEntityManager) ResolveChassis(lookup *service.EntityLookup, ccSerial string) (*service.ChassisEntity, error) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	ch, ok := m.chassisInventory[*chassis]
-	if !ok {
-		return nil, status.Errorf(codes.NotFound, "Could not find chassis with serial#: %s and manufacturer: %s", chassis.SerialNumber, chassis.Manufacturer)
+	chassis, found := m.chassisInventory[*lookup]
+	if !found {
+		if lookup.SerialNumber == "" && ccSerial != "" {
+			ch, err := m.resolveChassisViaControllerCard(lookup, ccSerial)
+			if err != nil {
+				return nil, status.Errorf(codes.NotFound, "Could not find chassis with serial#: %s and manufacturer: %s and controller card %s",
+					lookup.SerialNumber, lookup.Manufacturer, ccSerial)
+			}
+			chassis = ch
+		} else {
+			return nil, status.Errorf(codes.NotFound, "Could not find chassis with serial#: %s and manufacturer: %s and controller card %s",
+				lookup.SerialNumber, lookup.Manufacturer, ccSerial)
+		}
+	}
+	return &service.ChassisEntity{BootMode: chassis.GetBootMode()}, nil
+}
+
+// find controller card
```

Comment:
Comments should be full sentences and should always be capitalized and punctuated.
i.e. // Find controller card.